pointer-masking-proposal.adoc

```diff
@@ -1,3 +1,370 @@
-# *RISC-V Pointer Masking Proposal*
+[[risc-v-pointer-masking-proposal]]
+# RISC-V Pointer Masking proposal
 
-(Placeholder)
+_Adam Zabrocki_, _Lee Campbell_, _Martin Maas_, _RISC-V TEE_ and _J Extension Task Groups_
+
+**Contributors: ** [Members of the TEE and J Extension Task Groups]
+
+[[introduction]]
+## Introduction
+
+RISC-V Pointer Masking (PM) is a feature which provides a possibility of implementing “memory tagging” by ignoring various bits (defined by mask) of the [.underline]#_effective address_# (virtual or physical) on RV128, RV64 and RV32.
+
+[.underline]#_Effective address_# is an address generated by the address generation logic before it is sent to the memory subsystem. As such, there is no special handling of physical vs. virtual memory at the software level.
+
+Memory tagging (MT) is a technique which can significantly improve the memory safety state of any application (including code running in more privileged modes). MT can be fully implemented in the hardware or partially hardware assisted. However, both implementations require a PM feature to be enabled to not overwhelm the performance.
+
+General idea of MT is based on assigning a tag to every granule of memory (a small, e.g. 16-byte, naturally aligned memory region). All accesses to memory must be made via a pointer with the correct tag. Use of an incorrect tag is noted and the monitor software can choose what to do with such a situation (e.g. could be killing the process, or just report it to the user immediately). To be able to store the tag in every memory pointer (poisoning the pointer) and correctly using it in running applications without significant performance impact, PM is necessary.
+
+This document proposes an extension for RISC-V called PM (Pointer Masking) which in the future might be enhanced for implementing a full hardware memory tagging.
+
+[[proposal]]
+## Proposal
+
+We propose to add:
+
+* new PM configuration CSR register called _**MMTE**_ (Memory Tagging Extension)
```

Comment:
Consolidate on "memory tag" vs. "pointer mask" terms?

Reply:
Right. It's interesting question. In the long term perspective, this proposal can be extended to full Memory Tagging, that's why we are using this name. However, we can discuss during the meeting if we should change that name